poutyne/framework/model.py

```diff
@@ -334,17 +355,30 @@ def fit_generator(self,
 
         return epoch_iterator.epoch_logs
 
-    def _fit_batch(self, x, y, *, callback=Callback(), step=None, return_pred=False):
-        self.optimizer.zero_grad()
+    def _fit_batch(self,
+                   x,
+                   y,
+                   *,
+                   callback=Callback(),
+                   step=None,
+                   return_pred=False,
+                   zero_all_gradients=True,
+                   do_backprop=True,
+                   batches_per_step=1):
+        if zero_all_gradients:
+            self.optimizer.zero_grad()
 
         loss_tensor, metrics, pred_y = self._compute_loss_and_metrics(x,
                                                                       y,
                                                                       return_loss_tensor=True,
                                                                       return_pred=return_pred)
 
+        loss_tensor = loss_tensor / batches_per_step
```

Comment:
do we need to validate a negative number ? or a too large batches per step value ?

Reply:
I know many libraries like Numpy and Scikit-learn do some kind of sanity check on all input arguments to provide their users with potentially more meaningful error messages. 

However, since this is not seen anywhere in Poutyne, I don't feel like it's needed to go beyond a clear description of the parameter's role, hence its possible values.

Reply:
Good with that.